### PR TITLE
Fixing build breaks due to non-CLS compliant code and proj files

### PR DESCRIPTION
**Pull request by sokket**
Fixing build breaks due to non-CLS compliant code and proj files

/cc @KrzysztofCwalina

### CODE REVIEW

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by sokket** [comment]
The first commit fixes the build breaks, I'm working on fixing the test failures now

**Comment by sokket** [comment]
Build and tests all pass now :)

File: src/System.Slices/tests/UsageScenarioTests.cs [hunk 1]
```diff
@@ -33,6 +33,7 @@ public MyByte(byte value)
         [InlineData(new byte[] { 0, 1, 2 })]
         [InlineData(new byte[] { 0, 1, 2, 3 })]
         [InlineData(new byte[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19 })]
+        [CLSCompliant(false)]
```
Comment (krwq) [inline src/System.Slices/tests/UsageScenarioTests.cs]: The weird fact is that this whole assembly has CLSCompliant(false) and compiler still complains about it. I guess this is ok workaround for now

Comment (krwq) [inline src/System.Slices/tests/UsageScenarioTests.cs]: Could you actually add TODO in here about that?

**Comment by krwq** [comment]
Thanks for fixing it! LGTM :+1: (please add TODO before merging)
I'm also wondering where the error about some missing file in System.Numerics come from (some template doesn't generate code when building from CMD). I saw this happening on two of my computers and on @ahsonkhan's computer.

File: src/System.Slices/tests/UsageScenarioTests.cs (hunk 1 above)
Comment (sokket) [inline src/System.Slices/tests/UsageScenarioTests.cs]: Done

**Comment by sokket** [comment]
Thanks @krwq! Will merge once the CI build passes

**Comment by sokket** [comment]
@dotnet-bot test this please

**Comment by krwq** [comment]
Thanks! :+1:

**Pull request merged**